addons/crm/models/crm_lead.py

```diff
@@ -1202,6 +1324,333 @@ def get_import_templates(self):
+                    # 2. Compute won and lost score using each variable's individual probability
+                    s_lead_won = p_won
+                    s_lead_lost = p_lost
+                    for variable, value in leads_values_dict[lead_id]:
+                        if vars_won_count[variable]:
+                            for key, val in vars_won_count[variable].items():
+                                if str(value) == key:
+                                    if variable == 'stage_id':
+                                        s_lead_won *= vars_won_count[variable][key] / won
+                                        s_lead_lost *= vars_lost_count[variable][key] / lost
+                                    else:
+                                        s_lead_won *= vars_won_count[variable][key] / vars_total_count[variable]['won']
+                                        s_lead_lost *= vars_lost_count[variable][key] / vars_total_count[variable]['lost']
+
+                    # 3. Compute Probability to win
+                    lead_probabilities[lead_id] = 100 * s_lead_won / (s_lead_won + s_lead_lost)
+        return lead_probabilities
+
+    def _update_frequency_table(self, values_to_create, stage_ids, stage_sequences, company, team_id=None):
+        """ Create / update the frequency table in a cross company way.
+        But as the PLS start date depends on the company, we still need to store the company id,
+        to compute the probability per company, to be limited only on the time range configured on the company."""
+        frequencies = {}
+        pls_start_date = company.predictive_lead_scoring_start_date.strftime('%Y-%m-%d')
+        team_condition = 'and l.team_id = %s' % int(team_id) if team_id else 'and l.team_id is null'
+
+        def _increment_frequency(field, value, won, lost):
+            if field not in frequencies:
+                frequencies[field] = {value: {'won': won, 'lost': lost}}
+            else:
+                if value not in frequencies[field]:
+                    frequencies[field][value] = {'won': won, 'lost': lost}
+                else:
+                    frequencies[field][value]['won'] += won
+                    frequencies[field][value]['lost'] += lost
+
+        # get all lead fields combination aggregated by won / lost count
+        fields = ['stage_id', 'team_id'] + company.predictive_lead_scoring_fields.mapped('name')
+        str_fields = ', '.join(field for field in fields)
+
+        sql = """select probability, active, %s, count(probability)
+                                    from crm_lead l
+                                    where (probability = 0 or probability >= 100)
+                                    and create_date > %%s
+                                    %s
+                                    group by probability, active, %s"""
+        self._cr.execute(sql % (str_fields, team_condition, str_fields), [pls_start_date])
```

Comment:
Again, sql formatting is dangerous (injections).
Should use the second argument of the execute to pass all parameters.

Reply:
Unfortunately, as I inject string, I cannot do another way. Unless you know a way to do that ? :) 
But what I can do is to check before injecting that each field exist in the table I will search on. (this is actually what the ORM is doing before each search). If I understood correctly what @rco-odoo told me.
-> See _get_pls_safe_fields

Reply:
Have you tried passing a tuple of strings containing your fields as the argument?
Never used that myself but it's how the ORM _should_ react to it so it's worth a try?